vmware-base/src/com/cloud/hypervisor/vmware/mo/VirtualMachineMO.java

```diff
@@ -932,6 +933,38 @@ else if (prop.getName().startsWith("value[")) {
         return networks;
     }
 
+    public List<DatastoreMO> getAllDatastores() throws Exception {
```

Comment:
This "Exception" here. Is it from a VMware JDK method?

Reply:
@rafaelweingartner Correct. VMware SDK exception.

Reply:
@sureshanaparti is it worth handling it locally and/or throwing a more specific exception up?

Reply:
IMHO, the code would benefit from a more specific exception.

Reply:
@GabrielBrascher it is not possible. The methods from VMware SDK are throwing `Exception`.